packages/typespec-client-generator-core/src/http.ts

```diff
@@ -338,88 +338,84 @@ export function getSdkHttpParameter(
-    if (isPathParam(context.program, param) || location === "path") {
-      // we don't url encode if the type can be assigned to url
-      const urlEncode = !ignoreDiagnostics(
-        program.checker.isTypeAssignableTo(
-          // TODO: THIS NEED TO BE MIGRATED BY MARCH 2024 release.
-          // eslint-disable-next-line @typescript-eslint/no-deprecated
-          param.type.projectionBase ?? param.type,
-          program.checker.getStdType("url"),
-          param.type,
-        ),
-      );
-      parameter = {
-        ...base,
-        kind: "path",
-        urlEncode,
-        explode: (httpParam as HttpOperationPathParameter)?.explode ?? false,
-        style: (httpParam as HttpOperationPathParameter)?.style ?? "simple",
-        allowReserved: (httpParam as HttpOperationPathParameter)?.allowReserved ?? false,
-        serializedName: getPathParamName(program, param) ?? base.name,
-        correspondingMethodParams,
-        optional: false,
-      };
-    } else if (isCookieParam(context.program, param) || location === "cookie") {
-      parameter = {
-        ...base,
-        kind: "cookie",
-        serializedName: getCookieParamOptions(program, param)?.name ?? base.name,
-        correspondingMethodParams,
-        optional: param.optional,
-      };
-    } else if (isBody(context.program, param) || location === "body") {
-      parameter = {
-        ...base,
-        kind: "body",
-        serializedName: param.name === "" ? "body" : getWireName(context, param),
-        contentTypes: ["application/json"], // will update when we get to the operation level
-        defaultContentType: "application/json", // will update when we get to the operation level
-        optional: param.optional,
-        correspondingMethodParams,
-      };
-    } else if (isQueryParam(context.program, param) || location === "query") {
-      parameter = {
-        ...base,
-        optional: param.optional,
-        collectionFormat: getCollectionFormat(context, param),
-        correspondingMethodParams,
-        kind: "query",
-        serializedName: getQueryParamName(program, param) ?? base.name,
-        explode: (httpParam as HttpOperationQueryParameter)?.explode,
-      };
-    } else {
-      if (!(isHeader(context.program, param) || location === "header")) {
-        diagnostics.add(
-          createDiagnostic({
-            code: "unexpected-http-param-type",
-            target: param,
-            format: {
-              paramName: param.name,
-              expectedType: "path, query, header, or body",
-              actualType: param.kind,
-            },
-          }),
-        );
-      }
-      parameter = {
-        ...base,
-        optional: param.optional,
-        collectionFormat: getCollectionFormat(context, param),
-        correspondingMethodParams,
-        kind: "header",
-        serializedName: getHeaderFieldName(program, param) ?? base.name,
-      };
-    }
-    context.__httpParameterCache.set(param, parameter);
```

Comment:
just curious why not just remove this line? Feel like the other changes are not quite necessary?

Reply:
just rollback to previous version.